tracer/src/Datadog.Trace.ClrProfiler.Native/cor_profiler.cpp

```diff
@@ -646,6 +687,217 @@ HRESULT STDMETHODCALLTYPE CorProfiler::ModuleLoadFinished(ModuleID module_id, HR
     {
         module_ids_.push_back(module_id);
 
+        // Scan module for [Trace] methods
+        mdTypeDef typeDef = mdTypeDefNil;
+        mdTypeRef typeRef = mdTypeRefNil;
+        static shared::WSTRING traceAttribute = WStr("Datadog.Trace.TraceAnnotations.TraceAttribute");
+        static LPCWSTR traceAttribute_cstring = traceAttribute.c_str();
+        bool foundType = false;
+
+        ComPtr<IUnknown> metadata_interfaces;
+        auto hr = this->info_->GetModuleMetaData(module_id, ofRead | ofWrite, IID_IMetaDataImport2,
+                                                 metadata_interfaces.GetAddressOf());
+
+        if (FAILED(hr))
+        {
+            Logger::Warn("ModuleLoadFinished failed to get metadata interface for ", module_id, " ",
+                            module_info.assembly.name);
+            return S_OK;
+        }
+
+        const auto& metadata_import = metadata_interfaces.As<IMetaDataImport2>(IID_IMetaDataImport);
+
+        hr = metadata_import->FindTypeDefByName(traceAttribute_cstring, mdTypeDefNil, &typeDef);
+        if (SUCCEEDED(hr))
+        {
+            foundType = true;
+            Logger::Debug("ModuleLoadFinished found the TypeDef for ", traceAttribute, " defined in Module ", module_info.assembly.name);
+        }
+        else
+        {
+            // Now we enumerate all type refs in this assembly to see if Datadog.Trace.Annotations.TraceAttribute is referenced
+            auto enumTypeRefs = Enumerator<mdTypeRef>(
+                [&metadata_import](HCORENUM* ptr, mdTypeRef arr[], ULONG max, ULONG* cnt) -> HRESULT {
+                    return metadata_import->EnumTypeRefs(ptr, arr, max, cnt);
+                },
+                [&metadata_import](HCORENUM ptr) -> void { metadata_import->CloseEnum(ptr); });
+
+            auto enumIterator = enumTypeRefs.begin();
+            while (enumIterator != enumTypeRefs.end())
+            {
+                mdTypeRef typeRef = *enumIterator;
+
+                // Check if the typeref matches
+                mdToken parent_token = mdTokenNil;
+                WCHAR type_name[kNameMaxSize]{};
+                DWORD type_name_len = 0;
+
+                hr = metadata_import->GetTypeRefProps(typeRef, &parent_token, type_name, kNameMaxSize,
+                                                      &type_name_len);
+
+                if (TypeNameMatchesTraceAttribute(type_name, type_name_len))
+                {
+                    foundType = true;
+                    Logger::Debug("ModuleLoadFinished found the TypeRef for ", traceAttribute,
+                                 " defined in Module ", module_info.assembly.name);
+                    break;
+                }
+
+                enumIterator = ++enumIterator;
+            }
+        }
+
+        // We have typeRef and it matches Datadog.Trace.Attribute
```

Comment:
```suggestion
        // We have typeRef and it matches Datadog.Trace.Annotations.TraceAttribute
```

Reply:
Fixed in 46ae5f8dd067fd13db2b590f9a283f7ed6c1edea

Reply:
Fixed in 46ae5f8dd067fd13db2b590f9a283f7ed6c1edea